src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -0,0 +1,974 @@
+        parameters_t{
+            {"add_value", std::to_string(addValue)},
+            {"sub_value", std::to_string(subValue)}});
+
+    Pipeline pipeline(*input_node, *output_node);
+    pipeline.connect(*input_node, *custom_node, {{pipelineInputName, customNodeInputName}});
+    pipeline.connect(*custom_node, *output_node, {{customNodeOutputName, pipelineOutputName}});
+
+    pipeline.push(std::move(input_node));
+    pipeline.push(std::move(custom_node));
+    pipeline.push(std::move(output_node));
+
+    ASSERT_EQ(pipeline.execute(), StatusCode::OK);
+    ASSERT_EQ(response.outputs().size(), 1);
+
+    this->checkResponse<float>(inputValues, [addValue, subValue](float value) -> float {
+        return value + addValue - subValue;
+    });
+}
+
+TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, SeriesOfCustomNodes) {
+    constexpr int N = 100;
+    constexpr int PARAMETERS = 2;
+    static_assert(PARAMETERS > 0);
+    static_assert(N > PARAMETERS);
+    static_assert((N % PARAMETERS) == 0);
+    // input      add-sub x N      output
+    //  O------->O->O...O->O------->O
+
+    const std::vector<float> inputValues{3.2, 5.7, -2.4};
+    this->prepareRequest(inputValues);
+
+    const std::array<float, PARAMETERS> addValue{1.5, -2.4};
+    const std::array<float, PARAMETERS> subValue{-5.1, 1.9};
+
+    auto input_node = std::make_unique<EntryNode>(&request);
+    auto output_node = std::make_unique<ExitNode>(&response);
+
+    std::unique_ptr<CustomNode> custom_nodes[N];
+    for (int i = 0; i < N; i++) {
+        custom_nodes[i] = std::make_unique<CustomNode>(customNodeName + std::to_string(i), library,
+            parameters_t{
+                {"add_value", std::to_string(addValue[i % PARAMETERS])},
+                {"sub_value", std::to_string(subValue[i % PARAMETERS])}});
+    }
+
+    Pipeline pipeline(*input_node, *output_node);
+    pipeline.connect(*input_node, *(custom_nodes[0]), {{pipelineInputName, customNodeInputName}});
+    pipeline.connect(*(custom_nodes[N - 1]), *output_node, {{customNodeOutputName, pipelineOutputName}});
+    for (int i = 0; i < N - 1; i++) {
+        pipeline.connect(*(custom_nodes[i]), *(custom_nodes[i + 1]), {{customNodeOutputName, customNodeInputName}});
+    }
+
+    pipeline.push(std::move(input_node));
+    pipeline.push(std::move(output_node));
+    for (auto& custom_node : custom_nodes) {
+        pipeline.push(std::move(custom_node));
+    }
+
+    ASSERT_EQ(pipeline.execute(), StatusCode::OK);
+    ASSERT_EQ(response.outputs().size(), 1);
+
+    this->checkResponse<float>(inputValues, [N, addValue, subValue](float value) -> float {
+        for (int i = 0; i < PARAMETERS; i++) {
+            value += (N / PARAMETERS) * addValue[i];
+            value -= (N / PARAMETERS) * subValue[i];
+        }
+        return value;
+    });
+}
+
+TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, ParallelCustomNodes) {
+    constexpr int N = 200;
+    constexpr int PARAMETERS = 5;
```

Comment:
Here PARAMETERS is a bit ambiguous -> does it relate to how many parameters each node take?

Maybe PARAMETERS_PAIRS_COUNT?

Reply:
Done